csrc/scheduler/normalization_inner.cpp

```diff
@@ -5,78 +5,146 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 // clang-format on
-#include <scheduler/reduction.h>
-
-#include <debug.h>
-#include <executor_utils.h>
-#include <grouped_reduction.h>
-#include <inlining.h>
+#include <ATen/cuda/CUDAContext.h>
 #include <instrumentation.h>
-#include <ir/all_nodes.h>
-#include <ir/iostream.h>
-#include <ir/utils.h>
-#include <options.h>
 #include <scheduler/debug_utils.h>
 #include <scheduler/normalization_inner.h>
-#include <scheduler/normalization_utils.h>
 #include <scheduler/reduction_utils.h>
-#include <scheduler/registry.h>
 #include <scheduler/registry_utils.h>
 #include <scheduler/utils.h>
-#include <scheduler/vectorize_helper.h>
-#include <transform_replay.h>
-
-#include <ATen/cuda/CUDAContext.h>
-
-#include <cmath>
 
 namespace nvfuser {
 
+constexpr auto schedule_heuristic = ScheduleHeuristic::InnerPersistent;
+
 InnerPersistentKernelScheduler::InnerPersistentKernelScheduler(
     Fusion* fusion,
     SchedulerRuntimeInfo& runtime_info,
     HeuristicSummary* data_cache)
-    : SchedulerEntry(ScheduleHeuristic::InnerPersistent) {
+    : SchedulerEntry(schedule_heuristic) {
   computeHeuristics(fusion, runtime_info, data_cache);
 }
 
 void InnerPersistentKernelScheduler::schedule(Fusion* fusion) {
-  FUSER_PERF_SCOPE("Schedule Persistent Fusion");
+  FUSER_PERF_SCOPE("Schedule InnerPersistent Fusion");
   scheduleInnerPersistentKernel(fusion, reductionParams());
 }
 
+namespace {
+
+bool checkReductionPattern(
+    Fusion* fusion,
+    const std::vector<TensorView*>& reduction_tvs) {
+  // Use root domain map to check the reduction ops have the same axes
+  FusionGuard fg(fusion);
+  ComputeAtRootDomainMap root_map;
+  root_map.build(true);
+
+  // check inner and outer reductions seperately
+  for (const auto it : c10::irange(1, reduction_tvs.size())) {
+    if (!registry_utils::checkPatternEquivalence(
+            reduction_tvs[it - 1], reduction_tvs[it], root_map)) {
+      scheduler_debug_utils::canScheduleRejectReason(
+          schedule_heuristic,
+          "unmapped reduction ",
+          reduction_tvs[it - 1],
+          " and ",
+          reduction_tvs[it]);
+      return false;
+    }
+  }
+  return true;
+}
+
+std::pair<int64_t, int64_t> getPersistentBufferSize(
+    Fusion* fusion,
+    SchedulerRuntimeInfo& runtime_info,
+    HeuristicSummary* data_cache,
+    const std::vector<TensorView*>& reduction_tvs) {
```

Comment:
(1) Transformed from a static function in the InnerPersistentKernelScheduler class into a utility function within an anonymous namespace.
(2) Removed checks that were unrelated to inner persistence.